diff --git a/Makefile b/Makefile
index 4ac2eb2..a34ee2a 100755
--- a/Makefile
+++ b/Makefile
@@ -71,17 +71,9 @@ build/ld:
 build/dso:
 	mkdir -p $@
 
-$(MUSL_ARCHIVE):
-	wget https://musl.libc.org/releases/$@
-
-musl: $(MUSL_ARCHIVE)
-	rm -rf musl.tmp
-	mkdir musl.tmp
-	tar xzf $(MUSL_ARCHIVE) -C musl.tmp --strip-components=1
-	mv musl.tmp musl
 
 musl/lib/libc.a: musl
-	cd musl && ./configure --disable-shared
+	cd musl && CFLAGS=-fPIC ./configure --disable-shared
 	+make -C musl
 
 $(TARGET_LD): $(OBJECTS_LD) musl/lib/libc.a
